core/tests/webdriverio_utils/ExplorationEditorMainTab.js

```diff
@@ -518,6 +518,12 @@ var ExplorationEditorMainTab = function() {
   this.expectContentToMatch = async function(richTextInstructions) {
     await waitFor.visibilityOf(
       stateContentDisplay, 'State content display not showing up');
+    // Wait for any text to be present to reduce flakes.
+    await waitFor.condition(
+      stateContentDisplay, async function() {
+        return await action.getText('Elem', stateContentDisplay);
```

Comment:
Is there some reason (conceptually) why the existing waitFor.visibilityOf() isn't sufficient here?

Reply:
Not a clue. I'm a bit afraid to dive into it, and really don't want to:) I don't know frontend well. 

Conceptually, we don't care much if the text doesn't appear at the same time as the container, we just care that it appears, right?

Reply:
Oh, I have a feeling I know why this is happening. I'm not sure which test is failing (could you please update the PR description?) but I think there was a bug introduced a while back where the exploration editor "loads" before showing a loading message.

The old behaviour was: there would be a "Loading..." message and when it disappeared, the exploration editor was fully loaded. But somehow along the way, the editor sometimes loads with "empty data" and a placeholder content before later changing to display the actual content. If the test is detecting the element but not the text in it, then I think that might be what is causing this issue.

We could either fix that, or do what you have here -- I'm OK with either since this flake is a bit too common atm. (Or maybe another approach is to do what you have here but add a TODO to fix the underlying problem.) If we go ahead with this change, could you make a specific function waitFor.contentToBeNonempty() instead? Thanks!

Reply:
It fixes [this](https://github.com/oppia/oppia/actions/runs/7644469454/job/20830554086) test, additionalEditorFeaturesModals. I'll update the description once I publish this, yeah.

Yeah, thanks, I'll leave a TODO. I could dive into fixing this bug, but I don't think the impact would be too significant.

> If we go ahead with this change, could you make a specific function waitFor.contentToBeNonempty() instead?

I don't think it will work. That would introduce a circular dependency: action.js -> waitFor.js -> action.js, because contentToBeNonEmpty would call action.getText

Reply:
Then can we instead change action.getText() to only return if nonempty (so the waitFor.condition is called from action.getText)? I think at the level of the page tests we should only be using high-level statements.

Reply:
Yeah, that sounds reasonable